hal/TARGET_FLASH_CMSIS_ALGO/flash_data.h

```diff
@@ -50,6 +50,9 @@ typedef struct {
  */
 struct flash_s {
     const flash_target_config_t *target_config;
+#if defined(__CORTEX_M23) || defined(__CORTEX_M33)
+    const flash_target_config_t *target_config_ns;
```

Comment:
@deepikabhavnani I expect the following behavior:
1. `flash_t` is initialized through secure gateway function `flash_init` for both secure/non-secure worlds, so all `target_config`/`target_config_ns`/`flash_algo` structs are initialized in secure-only `flash_set_target_config` (`flash_api.c`).
    ```
    struct flash_s {
        const flash_target_config_t *target_config;
    #if defined(__CORTEX_M23) || defined(__CORTEX_M33)
        const flash_target_config_t *target_config_ns;
    #endif
        const flash_algo_t *flash_algo;
    };
    ```
1. For non-secure world, `flash_t` should be seen as an opaque struct. It can be accessed only through secure gateway functions `flash_init`/`flash_free`/`flash_erase_sector` etc.  `flashIAP.h`/`flashIAP.cpp` follows this rule.